tendermint/src/lite/verifier.rs

```diff
@@ -154,3 +165,137 @@ where
 
     Ok(())
 }
+
+/// Returns Ok if we can trust the passed in (untrusted) header
+/// based on the given trusted state, otherwise returns an Error.
+fn can_trust<TS, SH, VS, L, S, R>(
+    trusted_state: &TS,    // h1 in spec
+    untrusted_header: &SH, // h2 in spec
+    trust_threshold: &L,
```

Comment:
> All of those things are expected to be immutable in the lifetime of a client.

Doesn't the store get modified/updated?

Also, aren't `trust_threshold` and `trusting_period` parameters which could change later. Let's say the user decides that going forward the client only accepts non-subsequent headers if 1/2 of the voting power signed without invalidating all trusted state of the past.

Reply:
I thought about putting trust_threshold, trusting_period into a struct called `TrustOptions` or `TrustParameters`. But didn't want to introduce more types just now.

Reply:
> and have this be a method on a Client that contains the req and store

Yes, with all these methods the client basically just contains req and store and we could pass the client instead of these two things separately. This means that the requester and the store of the client should be accessible for other functions by the client's public API (instead of sth that is passed in on initialization only).

Reply:
I mean the store variable itself is immutable in that it won't be swapped for another store (of course the underlying data it stores can change).

The trust parameters could change, but it's the kind of significant change that I think would warrant either a restart of a daemon or at least a re-init of the Client struct since it is in a sense a new client (with different approach to trust). You're right we don't want to invalidate past headers, so maybe we should be considering storing the trust_threshold with any headers we stored that weren't sequential, at least for reference. In any case I see no reason why the store couldn't be passed forward to a new client with new trust parameters (like it would be on a daemon restart with new params).

Seems a Client type that encapsulates these things is probably worth it to simplify the method args, but we can address it in a follow up.

Reply:
Added to #96